cmd/neofs-node/container.go

```diff
@@ -351,6 +354,82 @@ func setContainerNotificationParser(c *cfg, sTyp string, p event.NotificationPar
 	c.cfgContainer.parsers[typ] = p
 }
 
+func initPaymentChecker(c *cfg) {
+	var (
+		l = c.log.With(zap.String("component", "paymentChecker"))
+		p = &paymentChecker{
+			m:          sync.RWMutex{},
+			statuses:   make(map[cid.ID]int64),
+			balanceCli: c.bCli,
+		}
+	)
+
+	const changeUnpaidStatusEventName = "ChangePaymentStatus"
+	typ := event.TypeFromString(changeUnpaidStatusEventName)
+	c.cfgBalance.parsers[typ] = balanceEvent.ParseChangePaymentStatus
+	c.cfgBalance.subscribers[typ] = append(c.cfgBalance.subscribers[typ], event.WorkerPoolHandler(c.cfgBalance.workerPool, func(e event.Event) {
```

Comment:
Frankly, a pool of workers is too much for the thing they do (change some status in a map).